src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts

```diff
@@ -42,7 +40,7 @@ export class DotEditContentComponent implements OnInit {
     };
     dialogTitle: string;
     isModelUpdated = false;
-    page: DotRenderedPage;
+    pageState: DotRenderedPageState;
```

Comment:
Now we have a "BIG" main object to rule the page and the state and pass it into the container, it have everything we need, the page data, the page html and the page state, so on any change we just update that one, one source of truth.